.github/workflows/release.yml

```diff
@@ -17,6 +17,47 @@ defaults:
     shell: bash
 
 jobs:
+  shellcheck:
```

Comment:
it might make sense to move this to a different github actions script which only runs on pull requests